### ISSUE TITLE
Testing a pre-trained model

### ISSUE THREAD

**Issue by carlosfranzreb** [NVIDIA/NeMo#4655]
Hello,

I am trying to test a pre-trained conformer transducer, but the reported WER is around 100 %, whereas it is much lower when I use the `transcribe()` method of the model and compute the WER myself. What am I doing wrong?

Script:

```python
import yaml
from nemo.collections.asr.models.rnnt_bpe_models import EncDecRNNTBPEModel
import pytorch_lightning as pl

trainer = pl.Trainer(devices=1, accelerator="gpu", max_epochs=50)
config = yaml.load(open("./config/config.yaml"), Loader=yaml.FullLoader)
ct = EncDecRNNTBPEModel.from_pretrained("stt_de_conformer_transducer_large")

ct.setup_test_data(config["data"]["test_ds"])
trainer.test(ct, dataloaders=ct.test_dataloader())
```

Config file:
```yaml
data:
    test_ds:
    manifest_filepath: data/train_split/debug.txt
    sample_rate: 48000
    batch_size: 1
    shuffle: false
    num_workers: 0
    pin_memory: true
    use_start_end_token: false
```

**Comment by carlosfranzreb** [comment]
The WER achieved by using the `transcribe()` method is also inferior to the one reported in the [model's card](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/nemo/models/stt_de_conformer_transducer_large). I get an avg. WER of 18 % when evaluating the model on Common Voice 10.

I use this script to transcribe the samples and compute the WER, with the same config file as above.

```python
import json
import yaml
from nemo.collections.asr.models.rnnt_bpe_models import EncDecRNNTBPEModel
from nemo.collections.asr.metrics.wer import word_error_rate

config = yaml.load(open("./config/config.yaml"), Loader=yaml.FullLoader)
model = EncDecRNNTBPEModel.from_pretrained("stt_de_conformer_transducer_large")

wers, results = dict(), dict()
for file in config["data"]["test_ds"]["manifest_filepath"]:
    results[file] = dict()
    audiofiles, transcripts = [], []
    for line in open(file):
        obj = json.loads(line)
        audiofiles.append(obj["audio_filepath"])
        transcripts.append(obj["text"].lower())
    predicted = model.ct.transcribe(
        audiofiles,
        batch_size=config["data"]["test_ds"]["batch_size"],
        num_workers=config["data"]["test_ds"]["num_workers"],
    )[0]
    wers[file] = word_error_rate(predicted, transcripts)
json.dump(wers, open("avg_wers.json", "w"))
```

**Comment by titu1994** [comment]
You cannot pass sample rate 48000 to the rest dataloader - model was trained with just 16khz and so you need to remove that line, it will figure out how to resample the audio to 16khz. 

For the second example, two things. 1) this model was trained on MCV 7, not 10, so it is expected that wer will not match the model card. 

2) your method of calculating wer is wrong. WER is a global metric such as F1, not a metric that you compute per file then average. Compute all the predictions and all the ground truths then pass the full lists to wer once at the end. Or just use the transcribe_speech.py and speech_to_text_eval.py to evaluate your model which internally calls transcribe method anyway

**Comment by carlosfranzreb** [comment]
Thank you for your comments, @titu1994.

Changing the sample rate to 16khz does not affect the results. I understand that the model cannot be expected to perform as good for MCV 10 as for MCV 7, but the large difference in WER (over 3x) makes me think there is something wrong with my script.

I have looked at a couple of examples, and they sometimes differ only in their punctuation. Given that the conformer transducer does not output punctuation, should I remove it from the evaluation dataset? Is this how it was done in the results showed in the model card?

Regarding how I compute the WER, I am currently doing so per accent, i.e. each file contains samples of a different accent. I understand your point, but I am interested in the difference in performance per accent.

**Comment by titu1994** [comment]
We always normalize text before training, so yes it would be best to try to get text as close as possible to normalized text. 

We often use trainer.test() and it gives close enough results, so I don't know why in your case it is not working. We'd advise to use the scripts in the repo - speech to text eval and transcribe speech to do inference correctly

**Comment by carlosfranzreb** [comment]
Alright. Thank you very much!
